src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs

```diff
@@ -308,6 +308,12 @@ internal override void OnSelectedItemChanged(object oldSelectedItem, object sele
 				updateItemSelectedState: updateItemSelectedState);
 
 			UpdateContentPresenter();
+
+			if (updateItemSelectedState)
```

Comment:
A similar logic is found inside the `base.OnSelectedItemChanged()` where base is the `Selector` class, but it's behind  new flag `_changingSelectedIndex` https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs#L174 that was added to fix an issue with the `ListViewBase.cs` 

This change feels the least intrusive as it will only affect the `ComboBox`

Reply:
Btw: We are doing something similar from the `ListViewBase.cs` https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs#L303

Reply:
@ajpinedam thanks! You mentioned that it is a change that happened recently, could you point to the PR you found?

Reply:
@jeromelaban, this was the PR. https://github.com/unoplatform/uno/pull/12792

Reply:
btw: "recently" (in quotes). The change was made back in July but it was released in Uno 5.0.19

Reply:
@Youssef1313 this change may be of interest to you, as you worked on the original PR.

Reply:
Or, no. I think we should add a test for ComboBox selection when items are duplicated.

The problem is that when we try to select given only the item, not an index, we end up selecting the first matched item regardless of what the user actually clicked. So that case needs to be tested.